### ISSUE TITLE
Epub page numbers

### ISSUE THREAD

**Issue by kzshantonu** [koreader/koreader#4521]
Hi. Was wondering if there's a way to display page numbers embedded in the epub file, instead of number of page turns. Maybe I cannot find the option

**Comment by poire-z** [comment]
No, there's no option for that.
There is currently nothing to parse and show the various ways of having "absolulte"/fixed page numbers in an EPUB:
https://wiki.mobileread.com/wiki/Page_numbers
https://www.mobileread.com/forums/showthread.php?t=243222
(crengine only does _rendered page numbers_, and quite well.)

I guess it wouldn't be impossible to have crengine parse and expose these additional page thingies, but as there seem to be at least 3 ways of specifying them - and not one stands out - that would be quite some work.

**Comment by Frenzie** [comment]
Don't these two stand out?

https://wiki.mobileread.com/wiki/Adobe_Digital_Editions#Page-map

https://wiki.mobileread.com/wiki/NCX

Where technically NCX is standard.

I'll close this because it's unlikely to be implemented any time soon. Sorry for the disappointment.

**Comment by poire-z** [comment]
Looks like I have a few books with pageList in the toc.ncx.
Content looks like:
```xml
[... TOC entries ...]
  </navMap>
  <pageList class="pageList" id="pages">
    <pageTarget id="p1" value="I" type="front" class="noPage" playOrder="1">
      <navLabel>
        <text>I</text>
      </navLabel>
      <content src="cover.html"/>
    </pageTarget>
    <pageTarget id="p2" value="II" type="front" class="noPage" playOrder="2">
      <navLabel>
        <text>II</text>
      </navLabel>
      <content src="same001.html"/>
    </pageTarget>
    <pageTarget id="p3" value="III" type="front" class="noPage" playOrder="3">
      <navLabel>
        <text>III</text>
      </navLabel>
      <content src="tit001.html"/>
    </pageTarget>
    <pageTarget id="p4" value="7" type="normal" class="noPage" playOrder="5">
      <navLabel>
        <text>7</text>
      </navLabel>
      <content src="chap001.html#pg_7"/>
    </pageTarget>
    <pageTarget id="p5" value="8" type="normal" class="noPage" playOrder="6">
      <navLabel>
        <text>8</text>
      </navLabel>
      <content src="chap001.html#pg_8"/>
    </pageTarget>
```
And HTML content contains empty anchors like:
```html
particulière du chercheur, la
<a id="pg_9"/>mienne est différente.
```

And a few others with, in content.opf:
```xml
<item href="pageMap.xml" id="pageMap" media-type="application/oebps-page-map+xml"/>
<item href="toc.ncx" id="ncx" media-type="application/x-dtbncx+xml"/>
```
pageMap.xml contains:
```xml
<?xml version='1.0' encoding='UTF-8'?>
<page-map xmlns="http://www.idpf.org/2007/opf">
  <page href="meurtre-1.html" name="3"/>
  <page href="meurtre-2.html" name="5"/>
  <page href="meurtre-3.html" name="6"/>
  <page href="meurtre-4.html" name="7"/>
  <page href="meurtre-5.html" name="8"/>
  <page href="meurtre-6.html" name="9"/>
  <page href="meurtre-7.html" name="10"/>
  <page href="meurtre-8.html" name="11"/>
  <page href="meurtre-8.html#pg_12" name="12"/>
  <page href="meurtre-8.html#pg_13" name="13"/>
  <page href="meurtre-8.html#pg_14" name="14"/>
  <page href="meurtre-8.html#pg_15" name="15"/>
  <page href="meurtre-8.html#pg_16" name="16"/>
[...]
  <page href="meurtre-10.html#pg_172" name="172"/>
  <page href="meurtre-11.html" name="174"/>
</page-map>
```
With the same kind of empty anchors in the HTML.

Just for reference. (Not impossible to add to crengine, may be one day...)

**Comment by poire-z** [comment]
Implemented in #6004 and available in current nightly.
